onnxruntime/core/providers/openvino/backend_utils.cc

```diff
@@ -107,88 +98,100 @@ struct static_cast_int64 {
 
 std::shared_ptr<InferenceEngine::CNNNetwork>
 CreateCNNNetwork(const ONNX_NAMESPACE::ModelProto& model_proto, const GlobalContext& global_context, const SubGraphContext& subgraph_context, std::map<std::string, std::shared_ptr<ngraph::Node>>& const_outputs_map) {
-  std::shared_ptr<ngraph::Function> ng_function;
-  // NGraph Function
+ 
   if(IsCILogEnabled()) {
     std::cout << "CreateNgraphFunc" << std::endl;
   }
 
-
 #ifndef NDEBUG
   if (IsDebugEnabled()) {
     DumpOnnxModelProto(model_proto, subgraph_context.subgraph_name + "_static.onnx");
   }
 #endif
 
-#if (defined OPENVINO_2021_2) || (defined OPENVINO_2021_3)
-  ORT_UNUSED_PARAMETER(const_outputs_map);
-  std::istringstream model_stream{model_proto.SerializeAsString()};
-  try {
+  std::shared_ptr<ngraph::Function> ng_function;
+  #if (defined OPENVINO_2021_2) || (defined OPENVINO_2021_3)
+    ORT_UNUSED_PARAMETER(const_outputs_map);
+    std::istringstream model_stream{model_proto.SerializeAsString()};
+    try {
     ng_function = ngraph::onnx_import::import_onnx_model(model_stream);
     LOGS_DEFAULT(INFO) << "ONNX Import Done";
-  } catch (const std::exception& exp) {
-    ORT_THROW(log_tag + "[OpenVINO-EP] Exception while importing model to nGraph Func: " + std::string(exp.what()));
-  } catch (...) {
-    ORT_THROW(log_tag + "[OpenVINO-EP] Unknown exception while importing model to nGraph Func");
-  }
-#else
-  //ReadNetwork() API flow will be used in OpenVINO-EP starting from OpenVINO 2021.4
-  InferenceEngine::CNNNetwork cnn_network;
-  const std::string model = model_proto.SerializeAsString();
-  InferenceEngine::Blob::Ptr blob = {nullptr};
-  try {
-    cnn_network = global_context.ie_core.ReadNetwork(model, blob);
-    LOGS_DEFAULT(INFO) << "Read network Done";
-  } catch (const Exception& e) {
-    ORT_THROW(log_tag + "[OpenVINO-EP] Exception while Reading network: " + std::string(e.what()));
-  } catch (...) {
-    ORT_THROW(log_tag + "[OpenVINO-EP] Unknown exception while Reading network");
-  }
-  ng_function = cnn_network.getFunction();
-#endif
+    } catch (const std::exception& exp) {
+      ORT_THROW(log_tag + "[OpenVINO-EP] Exception while importing model to nGraph Func: " + std::string(exp.what()));
+    } catch (...) {
+      ORT_THROW(log_tag + "[OpenVINO-EP] Unknown exception while importing model to nGraph Func");
+    }
+  #elif defined (OPENVINO_2021_4)
+    const std::string model = model_proto.SerializeAsString();
+    auto cnn_network = global_context.ie_core.read_model(model);
```

Comment:
Shouldn't we use ReadNetwork() function from core API from OpenVINO 2021.4 and read_model() from 2022.1?
Will read_model() function (Which is from OV 2.0 API) work with OpenVINO 2021.4 as well?

Reply:
oh just realized, this is taken care in the read_model() method in ov_interface.cc